service/src/test/java/uk/gov/hmcts/reform/fpl/service/ActionCmoServiceTest.java

```diff
@@ -69,8 +118,59 @@ void shouldAddOrderActionToCaseDataAndCaseManagementOrderWhenApproved() {
     void shouldAddOrderActionToCaseDataButNotCaseManagementOrderWhenNotApproved() {
         CaseDetails caseDetails = CaseDetails.builder().data(new HashMap<>()).build();
 
-        service.prepareCaseDetailsForSubmission(caseDetails, CaseManagementOrder.builder().build(), false);
+        actionCmoService.prepareCaseDetailsForSubmission(caseDetails, CaseManagementOrder.builder().build(),
+            false);
 
         assertThat(caseDetails.getData()).containsOnlyKeys("orderAction");
     }
+
+    @Test
+    void shouldReturnEmptyStringWhenOrderActionIsNotPresentOnCMO() {
+        List<Element<HearingBooking>> hearingBookings = createHearingBookings(MOCK_DATE);
+        CaseManagementOrder caseManagementOrder = CaseManagementOrder.builder().build();
+
+        String label = actionCmoService.createNextHearingDateLabel(caseManagementOrder, hearingBookings);
+
+        assertThat(label).isEqualTo("");
+    }
+
+    @Test
+    void shouldFormatNextHearingBookingLabelWhenCMOOrderActionContainsMatchingUUID() {
+        List<Element<HearingBooking>> hearingBookings = createHearingBookings(MOCK_DATE);
+
+        CaseManagementOrder caseManagementOrder = CaseManagementOrder.builder()
+            .action(OrderAction.builder()
+                .nextHearingId(UUID.fromString("ecac3668-8fa6-4ba0-8894-2114601a3e31"))
```

Comment:
what do you think about moving this to a final variable at top?

Reply:
Since we used two different UUID's as nextHearingID I don't think we should. But we can import fromString static :)

Reply:
cool makes sense :)